mautic/mautic
# Fix for scenario where landing pages built with GrapesJS include unnecessary JS/CSS

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | features
| Bug fix?                               | yes
| New feature?                           | no
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | N/A
| Related user documentation PR URL      | N/A
| Related developer documentation PR URL | N/A
| Issue(s) addressed                     | Fixes #9998




#### Description:

As described in https://github.com/mautic/mautic/issues/9998#issuecomment-898915035, the reported issue is caused by JS and CSS tags having the `data-source="mautic"` attribute, which normally shouldn't be persisted to the database.

Before this change, HTML head JS/CSS tags with `data-source="mautic"` would incorrectly end up in the database:

![image](https://user-images.githubusercontent.com/17739158/130503980-114351ae-26a0-4438-8ae1-05fb512a0b0d.png)

After this change, HTML head JS/CSS tags don't end up in the database anymore, except the ones that are explicitly defined in the theme (`/nonexistent.js` in this case):

![image](https://user-images.githubusercontent.com/17739158/130504083-54b33bb2-a420-4020-923b-10866834198a.png)

For the actual fix, please refer to https://github.com/mautic/grapesjs-preset-mautic/pull/20.



#### Reproducing the bug:
1. Checkout the latest `features` branch
2. Update `themes/blank/html/base.html.twig` by adding `<script src="/nonexistent.js"></script>` somewhere in between the `<head></head>` tags (just before the closing `</head>` tag would be a good place
3. Open Mautic in developer mode: `https://mautic.ddev.site/index_dev.php`
4. Go to Components > Landing pages and create a new landing page.
5. Launch and close the builder, then save the landing page with a name you like
6. Go to https://mautic.ddev.site:8037/ in your browser, this opens phpMyAdmin. Find the `pages` table and click "edit" next to the page you just created
7. Find the `custom_html` column. See that all tags with the `data-source="mautic"` attribute are still there (see the first screenshot above)

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Repeat the steps above
3. Note that all tags with `data-source="mautic"` have correctly been stripped out before the page was persisted to the database (see the second screenshot above). Your custom `/nonexistent.js` from the theme should still be there.



**Important note when merging this PR**

This PR was created just so that end-users can test the fix that we created in https://github.com/mautic/grapesjs-preset-mautic/pull/20. The order of merging should be:
1. Merge https://github.com/mautic/grapesjs-preset-mautic/pull/20
2. Update the referenced Git commit by doing `cd plugins/GrapesJsBuilderBundle && npm update grapesjs-preset-mautic`
3. Run `npm run build` to generate the new dist files
4. Merge this PR
